### PR TITLE
[Tables] Cosmos workaround for Merge/Update

### LINKED ISSUE #13844: [Tables] Merge operation fails with a Cosmos endpoint
The Merge operation (which is implemented as Patch) always fails with a 405 MethodNotAllowed response against Cosmos table endpoints. Inspecting the traffic sent by the track 1 client reveals that they send Merge as Post with an additional header: X-HTTP-Method: MERGE . Adding this header and changing the method to Post results in a successful merge in my testing.

Ideally the Cosmos Table endpoint would accept the same http request shape as Storage.

#cosmosKnownIssue

[.NET Issue](https://github.com/Azure/azure-sdk-for-net/issues/13555)

### PR DESCRIPTION
**Pull request by seankane-msft**
Closes #13844

### CODE REVIEW

**Review by annatisch — COMMENTED** [review]
Looking good - thanks for this!

File: sdk/tables/azure-data-tables/tests/test_table_batch_cosmos.py [hunk 1]
```diff
@@ -279,8 +293,8 @@ def test_batch_update_if_match(self, tables_cosmos_account_name, tables_primary_
             batch = self.table.create_batch()
             batch.update_entity(
                 sent_entity,
-                etag=etag,
-                match_condition=MatchConditions.IfNotModified,
+                # etag=etag,
+                # match_condition=MatchConditions.IfNotModified,
```
[the remark this answers is not in the document]
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/tests/test_table_batch_cosmos.py]: Adding this back

File: sdk/tables/azure-data-tables/tests/test_table_batch_cosmos.py [hunk 2]
```diff
@@ -300,29 +314,37 @@ def test_batch_update_if_doesnt_match(self, tables_cosmos_account_name, tables_p
...
-            sent_entity1 = self._create_updated_entity_dict(entity['PartitionKey'], entity['RowKey'])
+            entity = self.table.get_entity(u'001', u'batch_update')
+            assert 3 ==  entity.test3
+            entity.test2 = u'value1'
 
             batch = self.table.create_batch()
-            batch.update_entity(
-                sent_entity1,
-                etag=u'W/"datetime\'2012-06-15T22%3A51%3A44.9662825Z\'"',
-                match_condition=MatchConditions.IfNotModified
```
[the remark this answers is not in the document]
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/tests/test_table_batch_cosmos.py:313]: Adding this too

**Review by annatisch — APPROVED** [review]
Looks good :)
Maybe add a note to the changelog?

**Pull request merged**